apis/tempo/v1alpha1/tempomonolithic_types.go

```diff
@@ -0,0 +1,269 @@
+const (
+	// MonolithicTracesStorageBackendMemory defines storing traces in a tmpfs (in-memory filesystem).
+	MonolithicTracesStorageBackendMemory MonolithicTracesStorageBackend = "memory"
+	// MonolithicTracesStorageBackendPV defines storing traces in a Persistent Volume.
+	MonolithicTracesStorageBackendPV MonolithicTracesStorageBackend = "pv"
+)
+
+// MonolithicTracesStorageWALSpec defines the write-ahead logging (WAL) configuration.
+type MonolithicTracesStorageWALSpec struct {
+	// Size defines the size of the Persistent Volume for storing the WAL. Defaults to 10Gi.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="10Gi"
+	Size resource.Quantity `json:"size"`
+}
+
+// MonolithicTracesStoragePVSpec defines the Persistent Volume configuration.
+type MonolithicTracesStoragePVSpec struct {
+	// Size defines the size of the Persistent Volume for storing the traces. Defaults to 10Gi.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="10Gi"
+	Size resource.Quantity `json:"size"`
+}
+
+// MonolithicIngestionSpec defines the ingestion settings.
+type MonolithicIngestionSpec struct {
+	// OTLP defines the ingestion configuration for OTLP
+	//
+	// +kubebuilder:validation:Optional
+	OTLP *MonolithicIngestionOTLPSpec `json:"otlp,omitempty"`
+
+	// TLS defines the TLS configuration for ingestion
+	//
+	// +kubebuilder:validation:Optional
+	TLS *MonolithicIngestionTLSSpec `json:"tls,omitempty"`
+}
+
+// MonolithicIngestionOTLPSpec defines the settings for OTLP ingestion.
+type MonolithicIngestionOTLPSpec struct {
+	// GRPC defines the OTLP/gRPC configuration
+	//
+	// +kubebuilder:validation:Optional
+	GRPC *MonolithicIngestionOTLPProtocolsGRPCSpec `json:"grpc,omitempty"`
+
+	// HTTP defines the OTLP/HTTP configuration
+	//
+	// +kubebuilder:validation:Optional
+	HTTP *MonolithicIngestionOTLPProtocolsHTTPSpec `json:"http,omitempty"`
+}
+
+// MonolithicIngestionOTLPProtocolsGRPCSpec defines the settings for OTLP ingestion over GRPC.
+type MonolithicIngestionOTLPProtocolsGRPCSpec struct {
+	// Enabled defines if OTLP over gRPC is enabled
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=true
+	Enabled bool `json:"enabled"`
+}
+
+// MonolithicIngestionOTLPProtocolsHTTPSpec defines the settings for OTLP ingestion over HTTP.
+type MonolithicIngestionOTLPProtocolsHTTPSpec struct {
+	// Enabled defines if OTLP over HTTP is enabled
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+}
+
+// MonolithicIngestionTLSSpec defines the TLS settings for ingestion.
+type MonolithicIngestionTLSSpec struct {
+	// Enabled defines if TLS is enabled for ingestion
+	//
+	// +kubebuilder:validation:Required
+	Enabled bool `json:"enabled"`
+
+	// CA defines the name of a secret containing the CA certificate
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	CA string `json:"ca"`
+
+	// Cert defines the name of a secret containing the TLS certificate and private key
```

Comment:
what are the entries in the secret?

Reply:
Sorry, I thought I removed this struct already. I'll use a generic TLS struct from TempoStack in the next PR where I add TLS for ingestion.